odl/util/utility.py

```diff
@@ -147,3 +128,89 @@ def __new__(cls, name, this_bases, d):
                 return type.__new__(cls, name, (), d)
             return meta(name, bases, d)
     return metaclass('temporary_class', None, {})
+
+
+def is_scalar_dtype(dtype):
+    """Whether a datatype is scalar or not."""
+    return np.issubsctype(dtype, np.number)
+
+
+def is_int_dtype(dtype):
```

Comment:
Maybe it's just me, but I have a hard time remembering which NumPy types to insert into which function. The nice thing about these helpers is that they have very specialized names. What's bad about them is that they are not necessarily as general as they pretend.

Another point is that they have not always been that uniform. For example, the `is_real_dtype` used to create this mock array and ask if it's real or not. This particular function still does slightly more than just wrapping one NumPy function. All the others are simple wrappers.

Maybe using the NumPy directly is more explicit and removes the danger of people using this for other than NumPy data types which they're clearly not intended for.